great_expectations/agent/agent.py

```diff
@@ -194,15 +194,52 @@ def _reject_correlation_id(self, id: str):
         return should_reject
 
     @classmethod
-    def _get_config_from_env(cls) -> GXAgentConfig:
-        """Construct GXAgentConfig from available environment variables"""
-        # TODO: replace with connection string from agent-sessions endpoint
-        url = os.environ.get("BROKER_URL", None)
-        org_id = os.environ.get("GX_CLOUD_ORGANIZATION_ID", None)
+    def _get_config(cls) -> GXAgentConfig:
+        """Construct GXAgentConfig."""
+
+        # ensure we have all required env variables, and provide a useful error if not
+
+        required_env_vars_with_defaults = {
+            "GX_CLOUD_BASE_URL": "https://api.greatexpectations.io",
+            "GX_CLOUD_ORGANIZATION_ID": None,
+            "GX_CLOUD_ACCESS_TOKEN": None,
+        }
+        for key, default_value in required_env_vars_with_defaults.items():
+            required_env_vars_with_defaults[key] = os.environ.get(key, default_value)
+        missing_vars = [
+            key for key, val in required_env_vars_with_defaults.items() if val is None
+        ]
+        if len(missing_vars):
+            NEWLINE = "\n    - "
+            missing_vars_str = NEWLINE + NEWLINE.join(v for v in missing_vars)
+            raise GXAgentError(
+                f"Missing or badly formed environment variables: {missing_vars_str}"
+            )
+
+        cloud_base_url = required_env_vars_with_defaults["GX_CLOUD_BASE_URL"]
+        organization_id = required_env_vars_with_defaults["GX_CLOUD_ORGANIZATION_ID"]
+        access_token = required_env_vars_with_defaults["GX_CLOUD_ACCESS_TOKEN"]
```

Comment:
You could defer all this logic to pydantic. 

https://docs.pydantic.dev/latest/usage/settings/

Reply:
🤯 so cool, thanks @Kilo59!

Reply:
updated in [1d20718](https://github.com/great-expectations/great_expectations/pull/8006/commits/1d207188f5eb47510cdc5558e90712dce0d0c9e7)